app/code/Magento/Inventory/Test/Integration/Stock/IsProductInStockTest.php

```diff
@@ -50,81 +27,36 @@ protected function setUp()
     {
         $this->indexer = Bootstrap::getObjectManager()->create(IndexerInterface::class);
         $this->indexer->load(StockItemIndexerInterface::INDEXER_ID);
-
-        $this->reservationBuilder = Bootstrap::getObjectManager()->get(ReservationBuilderInterface::class);
-        $this->reservationsAppend = Bootstrap::getObjectManager()->get(ReservationsAppendInterface::class);
-        $this->reservationCleanup = Bootstrap::getObjectManager()->get(ReservationCleanupInterface::class);
-
-        $this->isProductInStock = Bootstrap::getObjectManager()->get(IsProductInStockInterface::class);
+        $this->isProductInStock = Bootstrap::getObjectManager()->create(IsProductInStockInterface::class);
     }
 
     public function tearDown()
     {
-        /** @var IndexNameBuilder $indexNameBuilder */
-        $indexNameBuilder = Bootstrap::getObjectManager()->get(IndexNameBuilder::class);
-        /** @var IndexStructureInterface $indexStructure */
-        $indexStructure = Bootstrap::getObjectManager()->get(IndexStructureInterface::class);
-
-        foreach ([10, 20, 30] as $stockId) {
-            $indexName = $indexNameBuilder
-                ->setIndexId(StockItemIndexerInterface::INDEXER_ID)
-                ->addDimension('stock_', $stockId)
-                ->setAlias(Alias::ALIAS_MAIN)
-                ->build();
-            $indexStructure->delete($indexName, ResourceConnection::DEFAULT_CONNECTION);
-        }
-
-        // Cleanup reservations
-        $this->reservationCleanup->execute();
     }
 
     /**
      * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
      * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
-     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
      * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
-     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_link.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/reservations.php
```

Comment:
Need to revert stock_source_link because we need take into account indexer part also

Reply:
As far as I understand, the role of the `stock_source_link.php` fixture is that of assigning sources to stock but since `IsProductInStockTest.php` deals only with default source (which is already bound to default stock) I thought we could avoid using that fixture.

I don't get the point about the indexer, can you explain in which way it is involved in this test?

Reply:
> which is already bound to default stock

There is the same problem, Tests have to be green without InventoryCatalog
For example, you can remove `InventoryCatalog` but test need to be green
But now we have hard dependency on Default Stock (which has been created in InventoryCatalog)